ably/ablytest/recorders.go

```diff
@@ -422,7 +422,7 @@ type HostRecorder struct {
 
 	mu         sync.Mutex
 	httpClient *http.Client
-	dialWS     func(string, *url.URL) (proto.Conn, error)
+	dialWS     func(string, *url.URL, time.Duration) (proto.Conn, error)
```

Comment:
Suggestion: How about taking a `context.Context` instead? It's the established way of passing a timeout or deadline.

Reply:
We can't, the underlying `x/websocket` library uses `Dial` instead of `DialContext` to connect. So the only way to have the timeout is is through `Dialer` configuration.